src/containers/StructurePage/resourceComponents/AddResourceModal.tsx

```diff
@@ -40,32 +43,73 @@ const StyledContent = styled.div`
     background-color: white;
   }
 `;
-class AddResourceModal extends Component {
-  constructor() {
-    super();
+
+interface Props {
+  onClose: () => void;
+  type: string;
+  allowPaste: boolean;
+  topicId: string;
+  topicFilters: Filter[];
+  refreshResources: () => void;
+}
+
+interface ContentType {
+  id: string;
+  metaDescription?: string;
+  title?: string;
+  imageUrl?: string;
+}
+
+interface SelectedType {
+  id: string;
+  paths?: string[];
+  title?: string;
+  url?: string;
+  metaUrl?: string;
+  description?: string;
+  coverPhotoUrl?: string;
+}
+
+interface State {
+  selected: SelectedType | null;
+  content: ContentType | null;
+  pastedUrl: string;
+  error: string | undefined | null;
+  loading: boolean;
+}
+
+type SummaryTypes = LearningPathSearchSummary | GroupSearchSummary;
+
+class AddResourceModal extends Component<Props & tType, State> {
```

Comment:
Tror kanskje at når du først er i gang så kan du også gjøre om Class Component om til en Function Component. Da får vi bruk i litt nyere syntaks (hooks) og det er generelt den retninga hele prosjektet går i, vi skriver ikke nye komponenter som klassekomponenter men heller function components så blir mer konsistent med resten av kodebasen.

Reply:
Kan du utdype fordelen med å skrive om denne hooks for meg?
Jeg personlig kunne fint skrevet en ny komponent som klassekomponent dersom jeg følte for det, men om det finnes gode argumenter for å aldri bruke klassekomponenter vil jeg lære dem :smile:

Reply:
Nei altså klassekomponenter og funksjonkomponenter er ekvivalente og kan gjøre akkurat de samme tingene (uten om en ting, Error Boundaries, det kan kun skrives som klassekomponenter), så det har ikke så mye å gjøre med funksjonaliteten men med konsistens. Som beskrevet over så skriver vi nye filer i form av funksjonskomponenter og blir litt lettere å navigere og gjenbruke koden dersom alt er skrevet på samme format.

Reply:
Skrev den om til en functional component nå.
Jeg må innrømme at jeg ikke ser helt verdien, men nå er det i alle fall gjort :smile: 
Ta gjerne en ny titt :eyes: